docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,392 @@
+Distributed checkpoints
+=======================
+
+This guide provides details about the distributed checkpoints format from Megatron-Core.
+
+
+Introduction
+------------
+
+Model parallel training requires parallelism-aware checkpointing.
+Megatron-Core provides a checkpointing library capable of handling all types of parallelisms used in LLMs training.
+Although distributed checkpointing library is targeted at Megatron-Core model, it can be used with other models as well, provided an appropriate integration.
+
+The library provides two main entrypoints: `dist_checkpointing.save` and `dist_checkpointing.load` which are meant to replace the `torch.save` and `torch.load` in the regular checkpointing flow.
+Apart from that it provides mechanism to define different types of local tensors placement in the global checkpoint.
+
+
+Basic sharding
+--------------
+
+The main way to define relationship of a plain local PyTorch tensor to tensors on other ranks is by wrapping it in a `ShardedTensor` class.
+This allows to express the fact that a given local tensor is part of a larger *grid* of tensors of a given shape at a given offset.
+Instead of saving a simple state dict with `torch.Tensor`s, we save a *sharded* state dict with `dist_checkpointing.ShardedTensor`s.
```

Comment:
To establish the relationship between a plain local PyTorch tensor and tensors on other ranks, we wrap it in a ShardedTensor class. This approach allows us to express that a specific local tensor is part of a larger grid of tensors with a given shape and offset. Instead of saving a basic state dictionary with torch.Tensors, we save a sharded state dictionary with dist_checkpointing.ShardedTensors.